core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -188,7 +188,11 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
     BrokerTopicStats.TotalProduceRequestsPerSec -> MeterWrapper(BrokerTopicStats.TotalProduceRequestsPerSec, "requests"),
     BrokerTopicStats.TotalFetchRequestsPerSec -> MeterWrapper(BrokerTopicStats.TotalFetchRequestsPerSec, "requests"),
     BrokerTopicStats.FetchMessageConversionsPerSec -> MeterWrapper(BrokerTopicStats.FetchMessageConversionsPerSec, "requests"),
-    BrokerTopicStats.ProduceMessageConversionsPerSec -> MeterWrapper(BrokerTopicStats.ProduceMessageConversionsPerSec, "requests")
+    BrokerTopicStats.ProduceMessageConversionsPerSec -> MeterWrapper(BrokerTopicStats.ProduceMessageConversionsPerSec, "requests"),
+    BrokerTopicStats.NoKeyCompactedTopicRecordsPerSec -> MeterWrapper(BrokerTopicStats.NoKeyCompactedTopicRecordsPerSec, "requests"),
```

Comment:
@tuvtran @guozhangwang 
Since the decision is made to record the `InvalidRecord` errors only at `allTopicStats` level. Shall we move the L192-195 inside the `if (name.isEmpty)` check?

Reply:
@kamalcph Sorry for the late reply, I think the decision is to record it at specific-topic level, not on all-topics level.

Reply:
Going by the source-code, I can see the `NoKeyCompactedTopicRecordsPerSec`, `InvalidMagicNumberRecordsPerSec`, `InvalidMessageCrcRecordsPerSec` and `InvalidOffsetOrSequenceRecordsPerSec` metrics are marked only at the all-topic stats so I asked.